ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx

```diff
@@ -1,32 +1,49 @@
-import { Account } from 'Utils/propTypes'
 import { Dropdown, OverlayTrigger, SplitButton, Tooltip } from 'react-bootstrap'
-import { PropTypes as PM } from 'mobx-react'
+import { connectionStates } from '~/constants'
 import { Text } from 'react-localize'
 import { getJid, noop } from 'Utils'
 
 import React from 'react'
 import images from '~/images'
 
-const ConnectionSplitButtonProps = {
-  accounts: PM.observableArrayOf(Account).isRequired
-}
+export const EmptyDropdownItem = () => (<Dropdown.Item disabled><Text message='message.noAccountConfigured' /></Dropdown.Item>)
+
+export const DropdownItem = ({ isActive, jid, doChangeActiveAccount }) => (
+  <Dropdown.Item active={isActive} key={jid} eventKey={jid} onClick={isActive ? noop : () => doChangeActiveAccount(jid)}>{jid}</Dropdown.Item>
+)
+
+export const DropdownItems = ({ accounts, activeAccount, doChangeActiveAccount }) => (
+  <div>
+    {accounts.length <= 0 &&
+      <EmptyDropdownItem />
+    }
+    {accounts.map(getJid).sort().map(jid => (
+      <DropdownItem isActive={getJid(activeAccount) == jid} jid={jid} key={jid} doChangeActiveAccount={doChangeActiveAccount} />
+    ))}
+  </div>
+)
 
 class ConnectionSplitButton extends React.Component {
-  constructor (props) {
+  constructor(props) {
     super(props)
-    this.connectionStateMap = {
-      'INITIALIZING': { onClick: this.props.onConnect, messageId: 'action.connect', icon: images.accountDisconnectedIcon },
-      'NOT_CONNECTED': { onClick: this.props.onConnect, messageId: 'action.connect', icon: images.accountDisconnectedIcon },
-      'CONNECTED': { onClick: this.props.onDisconnect, messageId: 'action.disconnect', icon: images.accountConnectedIcon },
-      'ERROR': { onClick: noop, messageId: 'action.connectionError', icon: images.accountConnectionErrorIcon },
-      'CONNECTING': { onClick: noop, messageId: 'action.connecting', icon: images.accountConnectingIcon },
-      'DISCONNECTING': { onClick: noop, messageId: 'action.disconnecting', icon: images.accountDisconnectingIcon }
-    }
+    const onConnect = this.props.core.doConnect
+    const onDisconnect = this.props.core.doDisconnect
+
+    this.connectionStateMap = {}
+    this.connectionStateMap[connectionStates.INITIALIZING] = { onClick: onConnect, messageId: 'action.connect', icon: images.accountDisconnectedIcon }
```

Comment:
What is this state ? Looking at https://github.com/saros-project/saros/blob/master/core/src/saros/net/ConnectionState.java there is no such state.

Reply:
I think this state was introduced in order to define the state:
HTML-GUI is started, but no state was send/received.